### PR TITLE
percona-xtrabackup 8.0.32-26

### PR DESCRIPTION
**Pull request by jkrauska**
Created with `brew bump-formula-pr`.

Attempt at redoing https://github.com/Homebrew/homebrew-core/pull/131036

Two main issues:
- boost check patch needed updating
- use mysql [1] instead of mysql-client due to bad linker settings remain in mysql-client (full mysql makes more sense anyway as you need a server to use xtrabackup anyway)

1. mysql PR from a week back
https://github.com/Homebrew/homebrew-core/pull/131207
 

- [ ] `resource` blocks have been checked for updates.

license ref:

https://github.com/percona/percona-xtrabackup/#licensing
https://github.com/search?q=repo%3Apercona%2Fpercona-xtrabackup%20General%20Public%20License&type=code

### CODE REVIEW

**Comment by chenrui333** [comment]
thanks for taking on this :)

**Comment by jkrauska** [comment]
> thanks for taking on this :)

I'm new so this is tricky, but I'm interested in figuring it out.

**Comment by jkrauska** [comment]
Current failure: Can't link/include C library 'zlib', aborting.

File: Formula/percona-xtrabackup.rb
Comment (carlocab) [inline Formula/percona-xtrabackup.rb:92]: I don't think this patch is safe to remove. Does it no longer apply?

Comment (jkrauska) [reply Formula/percona-xtrabackup.rb:92]: will double check, but I believe applying the patch was failing. 

I removed so I could debug/find later bugs in the build which appear related to mysql changes which got in to mysql formula, but not mysql-client.
https://github.com/Homebrew/homebrew-core/pull/131207

Comment (jkrauska) [reply Formula/percona-xtrabackup.rb:92]: yes, as-is patch fails.  
BuildError: Failed executing: patch -g 0 -f -p1 -i /private/tmp/percona-xtrabackup--patch-20230526-39787-6cw12h/boost-check.patch

Comment (jkrauska) [reply Formula/percona-xtrabackup.rb:92]: it looks like the new format takes in to account different paths.

```
  GET_FILENAME_COMPONENT(HOMEBREW_BASE ${HOMEBREW_HOME} DIRECTORY)
```
only slightly different.

Variant is also used in mysql
https://raw.githubusercontent.com/Homebrew/formula-patches/030f7433e89376ffcff836bb68b3903ab90f9cdc/mysql/boost-check.patch

Comment (jkrauska) [reply Formula/percona-xtrabackup.rb]: Used newer version.

**Comment by jkrauska** [comment]
arm builds all worked.  x86s were taking longer.  let's see tomorrow! 🤞

**Comment by carlocab** [comment]
Linux is failing with
```
  /tmp/percona-xtrabackup-20230526-28878-iem1vs/percona-xtrabackup-8.0.32-26/storage/innobase/xtrabackup/src/utils.cc:26:10: fatal error: proc/sysinfo.h: No such file or directory
     26 | #include <proc/sysinfo.h>
        |          ^~~~~~~~~~~~~~~~
```
@Homebrew/linux?

**Comment by jkrauska** [comment]
All complete except for Linux.
Adding procps to linux requires wasn't helpful.

Percona currently provides MANY Linux binary packages on their download pages.
(Debian, RHEL, Centos, Ubuntu and "Generic")

Screenshot 2023-05-26 at 12 38 58 PM

Screenshot 2023-05-26 at 12 39 06 PM

It's MacOS builds which are completely missing and the major value I find with having this HomeBrew Formula.

Is it reasonable to disable Linux builds for this version?

**Comment by jkrauska** [comment]
Thread on percona forum suggesting procps-ng-devel is required.

https://forums.percona.com/t/arm-compile-error-percona-xtrabackup-80-8-0-32-26-1-generic-src-rpm/21534

**Comment by jkrauska** [comment]
Current procps package in homebrew in version 4.

This ticket on Percona's JIRA highlights the issue of compatibility with version 4 and lack of cmake check.

https://jira.percona.com/browse/PXB-2993?jql=text%20~%20%22sysinfo.h%22

will try patching.

**Comment by jkrauska** [comment]
@chenrui333 Do you have a position on the need for the Linux builds?

I think it might be possible to down-rev the procps formula and/or make an intentional procps-devel formula, but I don't know enough about the Linux use cases for Homebrew.

My struggle is that I don't have easy access to a Linux test platform so the test loop takes nearly an hour to run through GH CI, and that's wasteful.

**Comment by carlocab** [comment]
Which version of `procps` is needed?

**Comment by jkrauska** [comment]
> Which version of `procps` is needed? We can add a versioned formula for it if we need an older one.

The notes here: https://jira.percona.com/browse/PXB-2993?jql=text%20~%20%22sysinfo.h%22
Suggest that procps 3 is required.  

But note that it's the development headers (which I'm not sure are even installed in the normal package) that are needed here.  Not the lib itself.  (typically they would reside in a -devel package in rpm/deb speak).  I'm not sure what the equivalent is in brew and wasn't able to find any packages labeled '-devel'.

**Comment by jkrauska** [comment]
The equivalent debian package is:
https://packages.debian.org/bullseye/libprocps-dev

I can attempt to create a corresponding brew formulae, and then link it to this work.

**Comment by jkrauska** [comment]
Update: put together a Debian VM, so I could install HomeBrew and the procps formula.

It includes 
`./Cellar/procps/4.0.3/include/libproc2/misc.h`
Which looks like a promising include.

**Comment by jkrauska** [comment]
New error:
```
  /tmp/percona-xtrabackup-20230529-29318-6ls2d2/percona-xtrabackup-8.0.32-26/storage/innobase/xtrabackup/src/utils.cc: In function ‘long unsigned int xtrabackup::utils::host_total_memory()’:
  /tmp/percona-xtrabackup-20230529-29318-6ls2d2/percona-xtrabackup-8.0.32-26/storage/innobase/xtrabackup/src/utils.cc:144:3: error: ‘meminfo’ was not declared in this scope
    144 |   meminfo();
        |   ^~~~~~~
  /tmp/percona-xtrabackup-20230529-29318-6ls2d2/percona-xtrabackup-8.0.32-26/storage/innobase/xtrabackup/src/utils.cc:145:10: error: ‘kb_main_total’ was not declared in this scope
    145 |   return kb_main_total * 1024;
        |          ^~~~~~~~~~~~~
  /tmp/percona-xtrabackup-20230529-29318-6ls2d2/percona-xtrabackup-8.0.32-26/storage/innobase/xtrabackup/src/utils.cc: In function ‘long unsigned int xtrabackup::utils::host_free_memory()’:
  /tmp/percona-xtrabackup-20230529-29318-6ls2d2/percona-xtrabackup-8.0.32-26/storage/innobase/xtrabackup/src/utils.cc:149:3: error: ‘meminfo’ was not declared in this scope
    149 |   meminfo();
        |   ^~~~~~~
  /tmp/percona-xtrabackup-20230529-29318-6ls2d2/percona-xtrabackup-8.0.32-26/storage/innobase/xtrabackup/src/utils.cc:150:10: error: ‘kb_main_available’ was not declared in this scope
    150 |   return kb_main_available * 1024;
        |          ^~~~~~~~~~~~~~~~~
  make[2]: *** [storage/innobase/xtrabackup/src/CMakeFiles/xtrabackup.dir/build.make:443: storage/innobase/xtrabackup/src/CMakeFiles/xtrabackup.dir/utils.cc.o] Error 1
```

There's also a meminfo.h.

Check it vs misc.h for comparisons.

**Comment by jkrauska** [comment]
kb_main_total is only defined in procps version 3. (not 4)

**Comment by jkrauska** [comment]
It looks like I'd need a /new/ Formula procps@3.rb
If this completes, I'll use that approach (instead of overriding the current procps at 4)

File: Formula/procps@3.rb
Comment (carlocab) [inline Formula/procps@3.rb:11]: You need this to avoid failing `brew audit`:
```suggestion

  keg_only :versioned_formulae
```

File: Formula/percona-xtrabackup.rb
Comment (carlocab) [inline Formula/percona-xtrabackup.rb:32]: Do we still want this? `mysql-client` had the same fix as `mysql` applied recently.

File: Formula/procps@3.rb
Comment (jkrauska) [reply Formula/procps@3.rb]: thanks.  adding keg_only.

I thought bottle do was to list to target builds?

**Review by carlocab — COMMENTED** [review]
Thanks for the work you've put in so far! I cleaned up your branch a bit to get it ready to merge. I think we can do so as soon as CI is green.

File: Formula/percona-xtrabackup.rb
Comment (jkrauska) [reply Formula/percona-xtrabackup.rb:32]: I have no strong feelings.  The tool (percona xtrabackup) is used for backing up mysql server, so I feel like having it tied to the server package is a bit more "correct" logically.

Comment (jkrauska) [reply Formula/percona-xtrabackup.rb:32]: > Thanks for the work you've put in so far! I cleaned up your branch a bit to get it ready to merge. I think we can do so as soon as CI is green.

Thanks! I've learned a lot about how HomeBrew works in the process.  Was fun.

**Comment by jkrauska** [comment]
Looks like it wants an @4 alias.
```
cd Aliases
ln -s ../Formula/procps.rb procps@4
```

File: Formula/percona-xtrabackup.rb
Comment (carlocab) [reply Formula/percona-xtrabackup.rb:32]: > I have no strong feelings. The tool (percona xtrabackup) is used for backing up mysql server, so I feel like having it tied to the server package is a bit more "correct" logically.

Ok, fine with keeping this as-is then.

**Comment by carlocab** [comment]
> Looks like it wants an @4 alias.
> 
> ```
> cd Aliases
> ln -s ../Formula/procps.rb procps@4
> ```

Yep, done.

**Comment by carlocab** [comment]
My approval here won't unlock the merge since I pushed to this PR, but another maintainer should approve this soon.

Thanks for your work here, @jkrauska! This was very nice work.

**Review by SMillerDev — DISMISSED** [review]

**Review by chenrui333 — DISMISSED** [review]

**Review by carlocab — DISMISSED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request merged**
